src/MoBi.Core/Commands/SetParentPathCommand.cs

```diff
@@ -56,18 +56,22 @@ void updatePaths(ObjectPath objectPath)
          }
 
          //this takes care of all parameters in the structure as well as in the neighborhoods
-         var allReferencingPaths = spatialStructure.TopContainers
+         spatialStructure.TopContainers
             .Union(spatialStructure.Neighborhoods)
             .SelectMany(x => x.GetPathsReferencing(referencesPath))
-            .Select(x => x.path);
-
-         allReferencingPaths.Each(updatePaths);
+            .Select(x => x.path)
+            .Each(updatePaths);
 
          var allReferencingPathsInNeighborhood = spatialStructure.Neighborhoods
-            .SelectMany(x => new[] {x.FirstNeighborPath, x.SecondNeighborPath})
-            .Where(referencesPath);
+            .SelectMany(x => new[] {new {neighborhood = x, neighbor = x.FirstNeighborPath}, new {neighborhood = x, neighbor = x.SecondNeighborPath}})
+            .Where(x => referencesPath(x.neighbor));
+
 
-         allReferencingPathsInNeighborhood.Each(updatePaths);
+         allReferencingPathsInNeighborhood.Each(x =>
+         {
+            updatePaths(x.neighbor);
+            context.PublishEvent(new NeighborhoodChangedEvent(x.neighborhood));
```

Comment:
I refactored this to ensure that I am publishing the neighborhood changed event. This defacto allows for a refresh of the tree at the node level. Works well :)